### DISCUSSION TITLE
Drop Tool

### DISCUSSION THREAD

**Discussion by baristezel** [eclipse-sirius/sirius-components#3527] [Q&A]
I have a question about `Drop Tool`.

When i tried to do somethings in the EclipseCon2022 Sirius Web 101 video on Youtube to Drop Tool,  i took error about AQL interpreter. You can see the error message below.

An error has occurred with the expression 'aql:semanticSource': Couldn't find the 'semanticSource' variable

**Comment by sbegaudeau** [comment]
Hi,

We don't have any variable named `semanticSource` in Sirius Web, we only have `semanticEdgeSource` and `semanticEdgeTarget` used while creating edges. You can find an example of a drop tool [here](https://github.com/eclipse-sirius/sirius-web/blob/0b889ec075a94eaa655d2a528cf3ee61d6115f7c/packages/sirius-web/backend/sirius-web-papaya/src/main/java/org/eclipse/sirius/web/papaya/representations/componentdiagram/tools/ComponentDiagramDropToolProvider.java) if you want.

Regards,

Stephane Begaudeau
